vision_agent/agent/vision_agent.py

```diff
@@ -397,41 +379,15 @@ def chat_with_artifacts(
             raise ValueError("chat cannot be empty")
 
         if not artifacts:
-            artifacts = setup_artifacts()
-
-        # NOTE: each chat should have a dedicated code interpreter instance to avoid concurrency issues
-        code_interpreter = (
-            self.code_interpreter
-            if self.code_interpreter is not None
-            and not isinstance(self.code_interpreter, str)
-            else CodeInterpreterFactory.new_instance(
-                code_sandbox_runtime=self.code_interpreter,
-                remote_path=artifacts.remote_save_path.parent,
-            )
-        )
+            artifacts = Artifacts(self.cwd)
 
-        if code_interpreter.remote_path != artifacts.remote_save_path.parent:
-            raise ValueError(
-                f"Code interpreter remote path {code_interpreter.remote_path} does not match artifacts remote path {artifacts.remote_save_path.parent}"
-            )
-
-        with code_interpreter:
+        with CodeInterpreterFactory.new_instance(
+            code_sandbox_runtime=self.code_sandbox_runtime,
+            remote_path=self.cwd,
+        ) as code_interpreter:
             orig_chat = copy.deepcopy(chat)
             int_chat = copy.deepcopy(chat)
             last_user_message = chat[-1]
-            media_list = []
-            for chat_i in int_chat:
-                if "media" in chat_i:
-                    for media in chat_i["media"]:
-                        media = cast(str, media)
-                        artifacts.artifacts[Path(media).name] = open(media, "rb").read()
-
-                        media_remote_path = (
-                            Path(artifacts.remote_save_path.parent) / Path(media).name
-                        )
-                        chat_i["content"] += f" Media name {media_remote_path}"  # type: ignore
```

Comment:
This line needs to stay in here, or at least somehow attach this to the message content. This is so the agent knows where the media path is so when it executes code it loads the correct path. You can see it in the prompt examples here https://github.com/landing-ai/vision-agent/blob/main/vision_agent/agent/vision_agent_prompts.py#L43

Reply:
addressed